Libplanet/Net/Swarm.cs

```diff
@@ -1381,6 +1408,19 @@ CancellationToken cancellationToken
                 if (pong.AppProtocolVersion != _appProtocolVersion)
                 {
                     dealer.Dispose();
+
+                    DifferentProtocolVersionEventArgs args =
+                        new DifferentProtocolVersionEventArgs
+                        {
+                            ExpectedVersion = _appProtocolVersion,
+                            ActualVersion = pong.AppProtocolVersion,
+                        };
+
+                    if (DifferentVersionPeerDiscovered != null)
+                    {
+                        DifferentVersionPeerDiscovered(this, args);
```

Comment:
Currently game programmers (library users) seem to have no much control on this situation, like ignoring this peer or treating this peer as compatible.

To me, if this event callback cannot return nor decide anything, it doesn't look useful.  Nodes can be aware that there are some peers having different versions of protocol, but they don't have option to ignore/reject them.

Reply:
For now, I only thought of options to ignore or deny. I think we can make version compatible options later. How do you think about it?

Reply:
> For now, I only thought of options to ignore or deny.

What does “to deny” mean here?

Reply:
"deny" was a wrong word. Just "to ignore". A peer is not added to the peer list if the protocol version is different. And also, if a different version of peer sends peer delta, it is ignored.

Reply:
If we can only ignore them and there's no other option, when would `DifferentVersionPeerDiscovered` be used mostly?  Why would a game need to receive this event and what would it do for this event?

Reply:
For example, using `DifferentVersionPeerDiscovered` the game can notify users of things like version upgrades.

Reply:
Ah, make sense. 👌